addon-test-support/@ember/test-helpers/dom/focus.ts

```diff
@@ -39,51 +44,68 @@ function getClosestFocusable(
 /**
   @private
   @param {Element} element the element to trigger events on
+  @return {Promise<FocusRecord | Event | void>} resolves when settled
 */
 export function __focus__(
   element: HTMLElement | Element | Document | SVGElement
-): void {
-  let focusTarget = getClosestFocusable(element);
-
-  const previousFocusedElement =
-    document.activeElement &&
-    document.activeElement !== focusTarget &&
-    isFocusable(document.activeElement)
-      ? document.activeElement
-      : null;
-
-  // fire __blur__ manually with the null relatedTarget when the target is not focusable
-  // and there was a previously focused element
-  if (!focusTarget) {
-    if (previousFocusedElement) {
-      __blur__(previousFocusedElement, null);
-    }
-
-    return;
-  }
-
-  let browserIsNotFocused = document.hasFocus && !document.hasFocus();
-
-  // fire __blur__ manually with the correct relatedTarget when the browser is not
-  // already in focus and there was a previously focused element
-  if (previousFocusedElement && browserIsNotFocused) {
-    __blur__(previousFocusedElement, focusTarget);
-  }
-
-  // makes `document.activeElement` be `element`. If the browser is focused, it also fires a focus event
-  focusTarget.focus();
+): Promise<FocusRecord | Event | void> {
+  return Promise.resolve()
+    .then(() => {
+      let focusTarget = getClosestFocusable(element);
+
+      const previousFocusedElement =
+        document.activeElement &&
+        document.activeElement !== focusTarget &&
+        isFocusable(document.activeElement)
+          ? document.activeElement
+          : null;
+
+      // fire __blur__ manually with the null relatedTarget when the target is not focusable
+      // and there was a previously focused element
+      return !focusTarget && previousFocusedElement
+        ? __blur__(previousFocusedElement, null).then(() =>
+            Promise.resolve({ focusTarget, previousFocusedElement })
+          )
+        : Promise.resolve({ focusTarget, previousFocusedElement });
+    })
+    .then(({ focusTarget, previousFocusedElement }) => {
+      if (!focusTarget) {
+        throw new Error('There was a previously focused element');
```

Comment:
Throwing an error (and later `catch`ing) here in order to mimic the behaviour previously handled by a en `return` statement.